homeassistant/components/weather/__init__.py

```diff
@@ -173,23 +451,69 @@ def ozone(self) -> float | None:
 
     @property
     def visibility(self) -> float | None:
-        """Return the visibility in native units."""
+        """Return the visibility for backward compatibility.
+
+        Should not be set by integrations.
+        """
         return self._attr_visibility
 
     @property
-    def visibility_unit(self) -> str | None:
+    def native_visibility(self) -> float | None:
+        """Return the visibility in native units."""
+        return self._attr_native_visibility
+
+    @property
+    def native_visibility_unit(self) -> str | None:
         """Return the native unit of measurement for visibility."""
-        return self._attr_visibility_unit
+        return self._attr_native_visibility_unit
+
+    @property
+    def visibility_unit(self) -> str:
```

Comment:
Do like this for backwards compatibility:
```suggestion
        """Return the visibility unit for backward compatibility.
        Should not be set by integrations.
        """
        return self._attr_visibility_unit

    @property
    def _visibility_unit(self) -> str:
```